iothub/device/src/TransientFaultHandling/ExponentialBackoff.cs

```diff
@@ -98,10 +98,23 @@ public override ShouldRetry GetShouldRetry()
                 if (currentRetryCount < this.retryCount)
                 {
                     Random random = new Random();
-                    int num = (int)((Math.Pow(2.0, (double)currentRetryCount) - 1.0) * (double)random.Next((int)(this.deltaBackoff.TotalMilliseconds * 0.8), (int)(this.deltaBackoff.TotalMilliseconds * 1.2)));
-                    int num2 = (int)Math.Min(this.minBackoff.TotalMilliseconds + (double)num, this.maxBackoff.TotalMilliseconds);
-                    retryInterval = TimeSpan.FromMilliseconds((double)num2);
-                    return true;
+                    try
```

Comment:
The code is too complicated to read. How about `if(num2<0) return maxBackoff;`

Reply:
Overflow may loop back around to look like a normal positive value again, so we do need the checked statement to correctly implement the exponential backoff.

Reply:
Then don't convert it to int. 
```
double interval = Math.Pow(2.0, currentRetryCount) - 1.0) * random.Next((int)(this.deltaBackoff.TotalMilliseconds * 0.8), (int)(this.deltaBackoff.TotalMilliseconds * 1.2) + this.minBackoff.TotalMilliseconds;
if (interval > this.maxBackoff.TotalMilliseconds) {
   retryInterval = maxBackoff;
} else [
  retryInterval = TimeSpan.FromMilliseconds(interval);
}
```

Reply:
As Cristian mentioned above, double's are unnecessarily heavy here. We can keep everything as an int as long as we have the checked statement